application/src/main/java/org/thingsboard/server/service/queue/DefaultTbCoreConsumerService.java

```diff
@@ -640,13 +664,15 @@ private void forwardToDeviceActor(TransportToDeviceActorMsg toDeviceActorMsg, Tb
     }
 
     private void forwardToAppActor(UUID id, Optional<TbActorMsg> actorMsg, TbCallback callback) {
-        if (actorMsg.isPresent()) {
-            log.trace("[{}] Forwarding message to App Actor {}", id, actorMsg.get());
-            actorContext.tell(actorMsg.get());
-        }
+        actorMsg.ifPresent(tbActorMsg -> forwardToAppActor(id, tbActorMsg));
```

Comment:
It looks cool but costs a lot . lambdas with finals are always instantiated. it means more memory and more CPU usage. and higher temperature in the server room as well :)